DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -19,48 +21,92 @@
     public class ReportsController : Controller
     {
         private readonly IActivityService activityService;
+        private readonly ICourseCategoriesDataService courseCategoriesDataService;
+        private readonly ICourseDataService courseDataService;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserDataService userDataService;
 
-        public ReportsController(IActivityService activityService)
+        public ReportsController(
+            IActivityService activityService,
+            IJobGroupsDataService jobGroupsDataService,
+            IUserDataService userDataService,
+            ICourseDataService courseDataService,
+            ICourseCategoriesDataService courseCategoriesDataService
+        )
         {
             this.activityService = activityService;
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userDataService = userDataService;
+            this.courseDataService = courseDataService;
+            this.courseCategoriesDataService = courseCategoriesDataService;
         }
 
         public IActionResult Index()
         {
             var centreId = User.GetCentreId();
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            int? courseCategoryId = null;
+            var validatedCategoryId = adminUser.CategoryId == 0 ? courseCategoryId : adminUser.CategoryId;
 
             var filterData = new ActivityFilterData(
-                DateTime.Today.AddYears(-1),
+                DateTime.UtcNow.Date.AddYears(-1),
                 DateTime.UtcNow,
                 null,
-                null,
+                validatedCategoryId,
                 null,
                 ReportInterval.Months
             );
 
+            Response.Cookies.SetReportsFilterCookie(filterData, DateTime.UtcNow);
+
             var activity = activityService.GetFilteredActivity(centreId, filterData);
 
-            var model = new ReportsViewModel(activity);
+            var jobGroupName = filterData.JobGroupId.HasValue
+                ? jobGroupsDataService.GetJobGroupName(filterData.JobGroupId.Value)
+                : "All";
+            jobGroupName ??= "All";
+
+            var categoryName = filterData.CourseCategoryId.HasValue
+                ? courseCategoriesDataService.GetCourseCategoryName(filterData.CourseCategoryId.Value)
+                : "All";
+            categoryName ??= "All";
+
+            var courseNames = filterData.CustomisationId.HasValue
+                ? courseDataService.GetCourseNameAndApplication(filterData.CustomisationId.Value)
+                : null;
+            var courseNameString = courseNames?.CompositeName ?? "All";

```

Comment:
This is getting quite long - can we pull getting the names based on ids on filter data into a service?